compiler/haskell-ide-core/src/Development/IDE/Functions/Compile.hs

```diff
@@ -222,25 +222,42 @@ compileModule opt mod packageState uniqSupply deps pkgs tmr =
 runGhcSessionExcept
     :: IdeOptions
     -> Maybe ParsedModule
-    -> PackageDynFlags
+    -> HscEnv
     -> Ex.ExceptT e Ghc a
     -> Ex.ExceptT e IO a
 runGhcSessionExcept opts mbMod pkg m =
     Ex.ExceptT $ runGhcSession opts mbMod pkg $ Ex.runExceptT m
 
 
-getGhcDynFlags :: IdeOptions -> ParsedModule -> PackageDynFlags -> IO DynFlags
+getGhcDynFlags :: IdeOptions -> ParsedModule -> HscEnv -> IO DynFlags
 getGhcDynFlags opts mod pkg = runGhcSession opts (Just mod) pkg getSessionDynFlags
 
 -- | Evaluate a GHC session using a new environment constructed with
 -- the supplied options.
 runGhcSession
     :: IdeOptions
     -> Maybe ParsedModule
-    -> PackageDynFlags
+    -> HscEnv
     -> Ghc a
     -> IO a
-runGhcSession IdeOptions{..} = optRunGhcSession
+runGhcSession IdeOptions{..} modu env act = runGhcEnv env $ do
+    modifyDynFlags $ \x -> x{importPaths = maybe [] moduleImportPaths modu ++ importPaths x}
+    act
+
+
+moduleImportPaths :: GHC.ParsedModule -> [FilePath]
+moduleImportPaths pm =
+    maybe [] (\modRoot -> [modRoot]) mbModuleRoot
+  where
+    ms   = GHC.pm_mod_summary pm
+    file = GHC.ms_hspp_file ms
+    mod'  = GHC.ms_mod ms
+    rootPathDir  = takeDirectory file
+    rootModDir   = takeDirectory . moduleNameSlashes . GHC.moduleName $ mod'
+    mbModuleRoot
+        | rootModDir == "." = Just rootPathDir
+        | otherwise = dropTrailingPathSeparator <$> stripSuffix rootModDir rootPathDir
```

Comment:
Can this ever be `Nothing`? If not, we should probably change the type of the function to `ParsedModule -> FilePath`.

Reply:
Yep, its always Nothing at most, and I'm not very happy about stripSuffix either - that looks like a bug. But I'm deliberately not changing this code in this go.